input/otlp/logs.go

```diff
@@ -182,6 +182,16 @@ func (c *Consumer) convertLogRecord(
 				event.Network.Connection = modelpb.NetworkConnectionFromVTPool()
 			}
 			event.Network.Connection.Type = v.Str()
+		case attributeDataStreamDataset:
+			if event.DataStream == nil {
+				event.DataStream = modelpb.DataStreamFromVTPool()
```

Comment:
Will this respect OTLP Resource.attributes and LogRecord.attributes or only one? What about ScopeLogs.Scope.attributes?

Reply:
Good call. It was only reading LogRecord.attributes, but I've updated the PR to only Resource.attributes (only one). Looking at the code, we never read from ScopeLogs.Scope.attributes

Reply:
This is currently unspecified in https://github.com/elastic/opentelemetry-dev/blob/main/docs/design-decisions/ingest/routing.md and I think we need to clarify this.

From an OTel Collector perspective, I think if we're only to support one, Scope.attributes would actually match the best. Reason being is that it's typical for a related dataset to emit a distinct Scope, which may come from the same Resource as other instrumentation. For example you may have two scopes, `otelcol/hostmetricsreceiver/cpu` and `otelcol/hostmetricsreceiver/memory`, which should be indexed into separate data streams coming from the same resource (host). Similarly you could have logs from two different application SDKs (say nginx and custom java app) from the same host, but should end up in different data streams for easier anaylsis and more compact storage.

That said, we could support all 3 levels with some precedence rules (I would expect LogRecord.attributes > Scope.attributes > Resource.attributes)

@axw have you given this any thought?

Reply:
I'm not certain than any of them will cover all scenarios, e.g.:
 - resource attributes aren't appropriate for routing two different data sets produced by one data source, since resource describes the data source
 - instrumentation scope attributes aren't appropriate for routing data for two different service instances, since they would have identical instrumentation, just different resource attributes (e.g. `service.instance.id`)
 - I'm not sure that signal-level attributes are ever appropriate?

> That said, we could support all 3 levels with some precedence rules (I would expect LogRecord.attributes > Scope.attributes > Resource.attributes)

Sounds good, for consistency with the [attribute precedence](https://github.com/elastic/opentelemetry-dev/blob/main/docs/design-decisions/ingest/mapping.md#attribute-precedence) we've already documented (sans top-level fields).

Reply:
> support all 3 levels with some precedence rules (I would expect LogRecord.attributes > Scope.attributes > Resource.attributes)

done

Reply:
> That said, we could support all 3 levels with some precedence rules (I would expect LogRecord.attributes > Scope.attributes > Resource.attributes)

I'm revisiting the docs and I'm a little confused here. The attribute precedence described by @joshdover (which I implemented in the PR atm) appears to be in the opposite order to our documented vision.